### PR TITLE
Change App constructor

### PR DESCRIPTION
**Pull request by k-nasa**
Change App constructor
close: https://github.com/rustasync/tide/issues/190
## Description

- Change type name of App structure to State
- Implement App::with_state constructor
- Change App::new(()) to App::new()
- Fixed the part that is 'App :: new (())'

The changes not related to this PR are also included
- Remove unnecessary line  546a223
- Formatting README ab3b2d7

Open another PR for modification of the generics argument 'AppData' (change to State)

## Motivation and Context

See https://github.com/rustasync/tide/blob/master/rfcs/001-app-new.md 

## How Has This Been Tested?

Run ```cargo build``` and ```cargo test```.
There seems to be no problem because the example code works.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/rustasync/tide/blob/master/.github/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by secretfader** [comment]
Since this PR implements the RFC proposed by @yoshuawuyts, I'll let him make the call on any edits that need to be made, but this looks :100: to me.

**Comment by fairingrey** [comment]
Agree, it looks good! Thanks for your work.

File: examples/graphql.rs [hunk 1]
```diff
@@ -47,7 +47,7 @@ type Schema = juniper::RootNode<'static, Query, Mutation>;
 // `Deserialize`, so we use `Json` extractor to deserialize the request body.
 async fn handle_graphql(mut cx: Context<Data>) -> EndpointResult {
```
Comment (yoshuawuyts) [inline examples/graphql.rs]: I think these should probably also be switched over to `Context<State>`

**Comment by yoshuawuyts** [comment]
3 positive reviews are in, I say we go ahead and merge this!

**Pull request merged**